### PR TITLE
Creating new identity

### PR DESCRIPTION
**Pull request by MithunW**

### CODE REVIEW

**Review by rumeshmadhusanka — APPROVED** [review]

File: src/main/java/ClientHandler/ClientHandler.java [hunk 1]
```diff
@@ -80,9 +80,12 @@ public void resolveClientRequest(JSONObject jsonPayload) throws ServerException,
 
                 case ClientConstants.TYPE_WHO:
 
+
             }
-        } catch (IOException | ServerException | ParseException e) {
-            throw e;
+        } catch (IOException e) {
```
Comment (ThaminduR) [inline src/main/java/ClientHandler/ClientHandler.java]: Collapse catch blocks into one

File: src/main/java/Messaging/Messaging.java [hunk 2]
```diff
@@ -56,6 +57,24 @@ public static void respond(JSONObject obj, Socket socket) throws IOException {
         dataOutputStream.flush();
     }
 
+    public static void broadcastClients(JSONObject obj) throws IOException {
```
Comment (ThaminduR) [inline src/main/java/Messaging/Messaging.java]: remove these methods, if not in use

File: src/main/java/Server/Room.java [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
+import java.util.ArrayList;
+
 public class Room {
-    private String roomId;
-    private String owner;
-    private String serverId;
+    private final String roomId;
+    private final String owner;
+    private final String serverId;
+    private ArrayList<String> clientIdentity;
```
Comment (ThaminduR) [inline src/main/java/Server/Room.java]: add concurrent data structure? @rumeshmadhusanka

File: src/main/java/Server/ServerState.java [hunk 4]
```diff
@@ -37,9 +38,7 @@ private ServerState(){
     public static ServerState getServerState() {
         if (serverState == null) {
             synchronized (ServerState.class) {
-                if (serverState == null) {
-                    serverState = new ServerState();
-                }
+                serverState = new ServerState();
```
Comment (ThaminduR) [inline src/main/java/Server/ServerState.java]: don't remove double check

File: src/main/java/Server/Room.java (hunk 3 above)
Comment (rumeshmadhusanka) [reply src/main/java/Server/Room.java]: +1

File: src/main/java/ClientHandler/ClientHandler.java (hunk 1 above)
Comment (MithunW) [reply src/main/java/ClientHandler/ClientHandler.java]: done

File: src/main/java/Messaging/Messaging.java (hunk 2 above)
Comment (MithunW) [reply src/main/java/Messaging/Messaging.java]: done

File: src/main/java/Server/Room.java (hunk 3 above)
Comment (MithunW) [reply src/main/java/Server/Room.java]: solved

File: src/main/java/Server/ServerState.java (hunk 4 above)
Comment (MithunW) [reply src/main/java/Server/ServerState.java]: done

File: src/main/java/ClientHandler/ClientHandler.java [hunk 5]
```diff
@@ -33,38 +35,50 @@ public ClientHandler(Socket clientSocket) {
...
-            while (!quitFlag) {
+            while(!quitFlag) {
                 String line = scanner.nextLine();
                 logger.debug("Received: " + line);
                 resolveClientRequest(Messaging.jsonParseRequest(line));
             }
-        } catch (IOException | ParseException | ServerException e) {
-            logger.info(e.getMessage());
+        } catch (IOException | ParseException e) {
+            logger.debug("Exception occurred " + e);
```
Comment (ThaminduR) [inline src/main/java/ClientHandler/ClientHandler.java]:
```suggestion
        } catch (IOException | ParseException | ServerException e) {
            logger.info(e.getMessage());
```

File: src/main/java/ClientHandler/ClientHandler.java [hunk 6]
```diff
@@ -33,38 +35,50 @@ public ClientHandler(Socket clientSocket) {
...
         } finally {
             logger.info("Connection has ended for client: " + clientSocket.getInetAddress() + ":" + clientSocket.getPort());
         }
     }
```
Comment (ThaminduR) [inline src/main/java/ClientHandler/ClientHandler.java]:
```suggestion
    /**
     * Resolve a given request.
     *
     * @param jsonPayload -  Request as a JSONObject.
     */
```

File: src/main/java/ClientHandler/ClientHandler.java [hunk 7]
```diff
@@ -49,22 +51,38 @@ public void run() {
...
-
     /**
```
Comment (ThaminduR) [inline src/main/java/ClientHandler/ClientHandler.java]:
```suggestion

    /**
```

File: src/main/java/ClientHandler/ClientHandler.java [hunk 8]
```diff
@@ -55,16 +57,33 @@ public void run() {
      *
      * @param jsonPayload -  Request as a JSONObject.
      */
-    public void resolveClientRequest(JSONObject jsonPayload) throws ServerException, IOException, ParseException {
+    public void resolveClientRequest(JSONObject jsonPayload) {
```
Comment (ThaminduR) [inline src/main/java/ClientHandler/ClientHandler.java]:
```suggestion
    public void resolveClientRequest(JSONObject jsonPayload) throws ServerException, IOException, ParseException {
```

File: src/main/java/ClientHandler/ClientHandler.java [hunk 9]
```diff
@@ -77,36 +96,45 @@ public void resolveClientRequest(JSONObject jsonPayload) throws ServerException,
...
 
                 case ClientConstants.TYPE_QUIT:
+                    this.quitFlag = true;
 
                 case ClientConstants.TYPE_WHO:
 
             }
         } catch (IOException | ServerException | ParseException e) {
-            throw e;
+            logger.info(e.getMessage());
```
Comment (ThaminduR) [inline src/main/java/ClientHandler/ClientHandler.java]:
```suggestion
            throw e;
```

File: src/main/java/Server/ServerState.java [hunk 10]
```diff
@@ -62,12 +63,16 @@ public void initialize(String serverId, String serverConf) {
...
             }
+            //create main hall chatroom
+
+
             //TODO remove hardcoded Leader value
             this.currentLeader = new Leader("s1","localhost",5555);
 
         } catch (FileNotFoundException e) {
-            e.printStackTrace();
+            logger.debug("Exception occurred " + e);
```
Comment (ThaminduR) [inline src/main/java/Server/ServerState.java]:
```suggestion
            logger.debug(e.getMessage());
```

**Pull request merged**